### PR TITLE
[Merged by Bors] - feat: define composition of an `AEEqFun` with a (quasi)measure preserving function

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Function/AEEqFun.lean [hunk 1]
```diff
@@ -206,6 +206,69 @@ theorem induction_on₃ {α' β' : Type _} [MeasurableSpace α'] [TopologicalSpa
...
+
+section compQuasiMeasurePreserving
+
+variable [MeasurableSpace β] {ν : MeasureTheory.Measure β} {f : α → β}
+
+open MeasureTheory.Measure (QuasiMeasurePreserving)
+
+/-- Composition of an almost everywhere equal function and a quasi measure preserving function.
+
+Many lemmas about this definition assume that `f` is a measure preserving map. -/
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/AEEqFun.lean:221]: This is not really true, as you use another definition for composition with measure preserving maps. Maybe add a pointer here to this other definition?

File: Mathlib/MeasureTheory/Function/LpSeminorm.lean [hunk 2]
```diff
@@ -921,6 +921,24 @@ theorem memℒp_map_measure_iff (hg : AEStronglyMeasurable g (Measure.map f μ))
   simp [Memℒp, snorm_map_measure hg hf, hg.comp_aemeasurable hf, hg]
 #align measure_theory.mem_ℒp_map_measure_iff MeasureTheory.memℒp_map_measure_iff
 
+theorem Memℒp.comp_of_map (hg : Memℒp g p (Measure.map f μ)) (hf : AEMeasurable f μ) :
+    Memℒp (g ∘ f) p μ :=
+  (memℒp_map_measure_iff hg.aestronglyMeasurable hf).1 hg
+
+theorem snorm_comp_measurePreserving {ν : MeasureTheory.Measure β} (hg : AEStronglyMeasurable g ν)
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/LpSeminorm.lean:928]: Do we still need the overload `MeasureTheory.Measure` now that `Measure` has been removed from core? (Might require bumping to latest master).

File: Mathlib/MeasureTheory/Function/LpSpace.lean [hunk 3]
```diff
@@ -839,6 +839,56 @@ theorem Memℒp.norm_rpow {f : α → E} (hf : Memℒp f p μ) (hp_ne_zero : p 
...
+
+/-- Composition of an `L^p` function with a measure preserving function is an `L^p` function. -/
+def compMeasurePreserving (f : α → β) (hf : MeasurePreserving f μ μb) :
+    Lp E p μb →+ Lp E p μ where
+  toFun g := ⟨g.1.compMeasurePreserving f hf, g.1.compMeasurePreserving_mem_Lp g.2 hf⟩
+  map_zero' := rfl
+  map_add' := by rintro ⟨⟨_⟩, _⟩ ⟨⟨_⟩, _⟩; rfl
+
+@[simp]
+theorem compMeasurePresving_val (g : Lp E p μb) (hf : MeasurePreserving f μ μb) :
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Function/LpSpace.lean:863]:
```suggestion
theorem compMeasurePreserving_val (g : Lp E p μb) (hf : MeasurePreserving f μ μb) :
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/MeasureTheory/Function/LpSeminorm.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/MeasureTheory/Function/LpSeminorm.lean:928]: UPD: I merged the latest `master` but `_root_.Measure` is still there.

File: Mathlib/MeasureTheory/Function/AEEqFun.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/MeasureTheory/Function/AEEqFun.lean]: Done.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**